src/detect-engine-iponly.c

```diff
@@ -1023,23 +1008,22 @@ void IPOnlyMatchPacket(ThreadVars *tv,
         SCReturn;
 
     uint32_t u;
+    uint8_t bitarray;
     for (u = 0; u < src->size; u++) {
         SCLogDebug("And %"PRIu8" & %"PRIu8, src->array[u], dst->array[u]);
 
-        /* The final results will be at io_tctx */
-        io_tctx->sig_match_array[u] = dst->array[u] & src->array[u];
+        bitarray = dst->array[u] & src->array[u];
 
         /* We have to move the logic of the signature checking
          * to the main detect loop, in order to apply the
          * priority of actions (pass, drop, reject, alert) */
-        if (io_tctx->sig_match_array[u] != 0) {
+        if (bitarray) {
             /* We have a match :) Let's see from which signum's */
-            uint8_t bitarray = io_tctx->sig_match_array[u];
             uint8_t i = 0;
 
             for (; i < 8; i++, bitarray = bitarray >> 1) {
                 if (bitarray & 0x01) {
-                    Signature *s = de_ctx->sig_array[u * 8 + i];
+                    Signature *s = de_ctx->sig_array[io_ctx->sig_mapping[u * 8 + i]];
```

Comment:
I wonder if it makes sense to avoid this additional indirection. Perhaps we can just store the `Signature` pointers themselves. Takes twice the space, but gets rid of this double lookup.

Reply:
I don't see why not..  I'm not sure about the performance impact, but the result could be easier to read with out the extra lookup